packages/image_picker/image_picker_ios/test/image_picker_ios_test.dart

```diff
@@ -878,6 +891,188 @@ void main() {
+          maxHeight: 10.0,
+        ),
+      ));
+      await picker.getMedia(
+          options: MediaOptions(
+        allowMultiple: true,
+        imageOptions: ImageOptions.createAndValidate(
+          maxWidth: 10.0,
+          maxHeight: 20.0,
+        ),
+      ));
+      await picker.getMedia(
+          options: MediaOptions(
+        allowMultiple: true,
+        imageOptions: ImageOptions.createAndValidate(
+          maxWidth: 10.0,
+          imageQuality: 70,
+        ),
+      ));
+      await picker.getMedia(
+          options: MediaOptions(
+        allowMultiple: true,
+        imageOptions: ImageOptions.createAndValidate(
+          maxHeight: 10.0,
+          imageQuality: 70,
+        ),
+      ));
+      await picker.getMedia(
+          options: MediaOptions(
+        allowMultiple: true,
+        imageOptions: ImageOptions.createAndValidate(
+          maxWidth: 10.0,
+          maxHeight: 20.0,
+          imageQuality: 70,
+        ),
+      ));
+
+      expect(
+        log.calls,
+        <_LoggedMethodCall>[
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': null,
+            'maxHeight': null,
+            'imageQuality': null,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': 10.0,
+            'maxHeight': null,
+            'imageQuality': null,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': null,
+            'maxHeight': 10.0,
+            'imageQuality': null,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': 10.0,
+            'maxHeight': 20.0,
+            'imageQuality': null,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': 10.0,
+            'maxHeight': null,
+            'imageQuality': 70,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': null,
+            'maxHeight': 10.0,
+            'imageQuality': 70,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+          const _LoggedMethodCall('pickMedia', arguments: <String, dynamic>{
+            'maxWidth': 10.0,
+            'maxHeight': 20.0,
+            'imageQuality': 70,
+            'requestFullMetadata': true,
+            'allowMultiple': true
+          }),
+        ],
+      );
+    });
```

Comment:
Optional, and applies to most of these: the pattern of having a bunch of different calls and call assertions in the same test makes it hard to understand what is being tested; as I've touched these tests I've been trying to split them apart. Ideally each test should have only one call, and be named to indicate what that specific version of the call is testing.

Reply:
I'm going to make this a todo for myself and make a new pr for it.